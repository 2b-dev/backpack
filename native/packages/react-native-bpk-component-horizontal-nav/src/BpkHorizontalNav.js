/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2017 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  ScrollView,
  Platform,
  StyleSheet,
  ViewPropTypes,
} from 'react-native';
import React from 'react';
import PropTypes from 'prop-types';

const tokens = Platform.select({
  ios: () => require('bpk-tokens/tokens/ios/base.react.native.common.js'), // eslint-disable-line global-require
  android: () => require('bpk-tokens/tokens/android/base.react.native.common.js'), // eslint-disable-line global-require
})();

const styles = StyleSheet.create({
  horizontalNav: {
    borderColor: 'transparent',
    borderBottomColor: tokens.colorGray100,
    borderWidth: tokens.borderSizeSm,
    flexDirection: 'row',
  },
  spaceAround: {
    flex: 0,
    flexGrow: 1,
    justifyContent: 'space-around',
  },
});

const BpkHorizontalNav = (props) => {
  const {
    children,
    spaceAround,
    style,
    ...rest
  } = props;
  const navStyle = [styles.horizontalNav];
  if (spaceAround) {
    navStyle.push(styles.spaceAround);
  }
  return (
    <ScrollView
      alwaysBounceHorizontal={false}
      contentContainerStyle={[navStyle, style]}
      horizontal
      showsHorizontalScrollIndicator={false}
      {...rest}
    >
      { children }
    </ScrollView>
  );
};

BpkHorizontalNav.propTypes = {
  children: PropTypes.node.isRequired,
  spaceAround: PropTypes.bool,
  style: ViewPropTypes.style,
};

BpkHorizontalNav.defaultProps = {
  spaceAround: false,
  style: null,
};

export default BpkHorizontalNav;
