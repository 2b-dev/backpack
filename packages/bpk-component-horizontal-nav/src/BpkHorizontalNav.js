import React, { PropTypes } from 'react';

import './bpk-horizontal-nav.scss';

const BpkHorizontalNav = (props) => {
  const classNames = ['bpk-horizontal-nav'];
  const { children, className, ...rest } = props;

  if (className) { classNames.push(className); }

  return (
    <nav className={classNames.join(' ')} {...rest}>
      <ul className="bpk-horizontal-nav__list">{children}</ul>
    </nav>
  );
};

BpkHorizontalNav.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

BpkHorizontalNav.defaultProps = {
  className: null,
};

export default BpkHorizontalNav;

