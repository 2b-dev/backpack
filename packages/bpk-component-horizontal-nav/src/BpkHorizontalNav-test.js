import React from 'react';
import renderer from 'react-test-renderer';
import BpkHorizontalNav from './BpkHorizontalNav';

describe('BpkHorizontalNav', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkHorizontalNav>
        My nav content.
      </BpkHorizontalNav>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
