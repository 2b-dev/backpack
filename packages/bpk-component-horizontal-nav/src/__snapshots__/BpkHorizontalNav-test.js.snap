exports[`BpkHorizontalNav should render correctly 1`] = `
<nav
  className="bpk-horizontal-nav">
  <ul
    className="bpk-horizontal-nav__list">
    My nav content.
  </ul>
</nav>
`;
